src/main/java/org/example/Phone.java

```diff
@@ -1,37 +1,96 @@
 package org.example;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-interface Phone {
+public abstract class Phone {
+    public String color;
+    protected String material;
+    protected String IMEI;
 
-    public List<String> addContact(Integer contactCriteriaNumber, String phoneNumber, String firstName, String lastName);
+    public Phone(String IMEI, String material, String color) {
+        this.IMEI = IMEI;
+        this.material = material;
+        this.color = color;
 
-    void call(String phoneNumber);
-    String[] sendMessage(String s, String[] strings);
+    }
 
-    String[] getFirstMessage(String phoneNumber);
-    String [] getSecondMessage(String phoneNumber);
+    Integer BATTERY_LIFE = 24;
+    public List<String> contactList = new ArrayList<>();
+    List<String> callHistory = new ArrayList<>();
+    List<String> messageHistoryForSpecificPhoneNumber = new ArrayList<>();
+    public Map<String, List<String>> messageList = new HashMap<>();
 
-    void getContact(int i);
+    public void addContact(Contact contact) {
+        contactList.add(contact.toString());
```

Comment:
some minimal validation would be good there to not insert contacts without a name or bad phone number